Pulse.Core/Calls/ConnectionInfo.cs

```diff
@@ -1,3 +1,9 @@
 namespace Pulse.Core.Calls;
 
-internal record ConnectionInfo(string RemoteIPAddress, int MinPort, int MaxPort);
\ No newline at end of file
+internal record ConnectionInfo(
+    string IPAddress,
+    int MinPort,
+    int MaxPort,
+    byte[] PublicKey,
```

Comment:
Are you sure byte arrays play well with JSON deserialization?
I assume they will be serialized as JSON arrays of numbers by default, which is less efficient than passing them as ASCII strings, like I did in the server. In any case, either the server or the client has to change to conform to a common interface